collect.py

```diff
@@ -202,7 +202,19 @@ def screenshot(driver, file_path):
     image = Image.open(file_path)
     image.save(file_path)
 
-
+    
+def get_code(driver, file_path):
+    source = driver.page_source
+    f = open(file_path, "w", encoding="utf-8")
+    domtree = driver.find_elements_by_css_selector("*")
```

Comment:
Your code is only storing the tag names, it should store the entire DOM model.

Reply:
Maybe we could store the page source only and then parse it later to get the DOM tree.

Reply:
That is, we could store `document.documentElement.outerHTML` only.

Reply:
![](https://upload.wikimedia.org/wikipedia/commons/thumb/5/5a/DOM-model.svg/428px-DOM-model.svg.png)

isn't this how a dom model is supposed to be like

Reply:
Your code is only storing the tag names of the elements and is ignoring everything else. We have to store everything instead.

You can store `document.documentElement.outerHTML`, which will contain the entire html of the page and then we can parse it using a html library.